lib/Analysis/Allocation.cpp

```diff
@@ -94,22 +88,33 @@ SmallVector<unsigned> getRepShapeForCvtLayout(RankedTensorType srcTy,
   return repShape;
 }
 
-SmallVector<unsigned>
-getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
-                             unsigned &outVec) {
-  auto srcTy = op.getSrc().getType();
-  auto dstTy = op.getType();
-  return getScratchConfigForCvtLayout(srcTy, dstTy, inVec, outVec);
+// TODO: extend beyond scalars
+static SmallVector<unsigned> getRepShapeForAtomicRMW(triton::AtomicRMWOp op) {
+  SmallVector<unsigned> smemShape;
+  if (isa<RankedTensorType>(op.getPtr().getType())) {
+    // do nothing or just assert because shared memory is not used in tensor up
+    // to now
+  } else {
+    // need only bytes for scalar
+    // always vec = 1 and elemsPerThread = 1 for scalar?
+    smemShape.push_back(1);
+  }
+  return smemShape;
+}
+
+static SmallVector<unsigned> getRepShapeForAtomicCAS(triton::AtomicCASOp op) {
+  return SmallVector<unsigned>{1};
```

Comment:
I'm confused by the difference between this function and the one above.  That one returns {} or {1}, whereas this one returns {1}.  Why does RMW not need shared memory for RankedTensorType, whereas CAS does need shared memory?  (Can we write the answer down in a comment in the code?)

Reply:
ditto